File: src/apps/EventViewer/src/InputMonitoringAlertWindowController.m
#import "InputMonitoringAlertWindowController.h"
#import "KarabinerKit/KarabinerKit.h"

@interface InputMonitoringAlertWindowController ()

@property(weak) IBOutlet NSWindow* parentWindow;

@end

@implementation InputMonitoringAlertWindowController

- (void)show {
  NSRect frame = NSMakeRect(
      self.parentWindow.frame.origin.x + (self.parentWindow.frame.size.width / 2) - (self.window.frame.size.width / 2),
      self.parentWindow.frame.origin.y + (self.parentWindow.frame.size.height / 2) - (self.window.frame.size.height / 2),
      self.window.frame.size.width,
      self.window.frame.size.height);

  [self.window setFrame:frame display:NO];
  [self.parentWindow addChildWindow:self.window
                            ordered:NSWindowAbove];
}

- (IBAction)openSystemPreferencesSecurity:(id)sender {
  [NSApplication.sharedApplication miniaturizeAll:self];
  [NSWorkspace.sharedWorkspace openURL:[NSURL URLWithString:@"x-apple.systempreferences:com.apple.preference.security?Privacy"]];
}

@end

File: src/apps/EventViewer/src/InputMonitoringAlertWindowController.swift
import Cocoa

public class InputMonitoringAlertWindowController: NSWindowController {
    @IBOutlet var parentWindow: NSWindow!

    public func show() {
        guard let window = window else { return }

        let frame = NSMakeRect(
            parentWindow.frame.origin.x + (parentWindow.frame.size.width / 2) - (window.frame.size.width / 2),
            parentWindow.frame.origin.y + (parentWindow.frame.size.height / 2) - (window.frame.size.height / 2),
            window.frame.size.width,
            window.frame.size.height
        )

        window.setFrame(frame, display: false)
        parentWindow.addChildWindow(window, ordered: .above)
    }

    @IBAction func openSystemPreferencesSecurity(_: NSButton) {
        NSApplication.shared.miniaturizeAll(self)

        let url = URL(string: "x-apple.systempreferences:com.apple.preference.security?Privacy")!
        NSWorkspace.shared.open(url)
    }
}
